convertXmlToJson/ant/build.xml

```
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Common ANT build script for Nictiz NTS projects.
+    
+    # Use of this build script
+    
+    The normal use of this build script is to import it from another ANT build script.
+    
+    # Parameters
+    
+    To run it, the following parameters are expected:
+    - convert.to.json.file : The path to a file, which contains a comma separated list of XML fixtures to convert. If
+      this parameter is empty, the file doesn't exist or is empty, this is taken as a signal that no conversion needs
+      to be done.
```

But you create a `tempfile` in `build-multiple` and `build-single`, which is the process that is always used. How would one prevent this file from being created if you do not want the conversion?

---

`build-multiple` and `build-single` are the files that include this build file, they are not part of the "neutral" tool chain in the repo. So that's the entire point.

---

Which makes this option kind of hypothetical? Wouldn't it be more transparent to just create the tempfile here then? If I'm correct it would remain empty anyways if no conversion needs to be done by the rules that are defined by the process (json-fixture that does not exist referenced, but xml-variant exists).

If I'm correct you would end up in this situation only when you plan on creating the json-fixtures yourself, but haven't done so yet, but the build fails anyways? Which would serve as a reminder for you to create those fixtures I guess,

If you do not want the conversion to happen for some other reason, wouldn't it be more transparent to make it skip based on a property being set?

---

I don't quite follow. For _this_ build file, it's only relevant if there is input to convert or not. If the file is empty _or_ absent, nothing is done. What use is there to create the file anyway and leave it empty?

For the NTS build file, it is relevant to know if it should fail on missing JSON fixtures or write them down for autoconversion. This is done by passing it a file to write the paths of fixtures to convert.

The consumers of these build files, that is `build-single` and `build-multiple`, include the step of autoconversion, so they define the file to write the fixture paths to. If you would want a flag to skip autoconversion, it should be defined in these files.

---

To the user of NTS in the Nictiz-testscripts, there is a single tool chain. This is built up from multiple back-ends, but that's irrelevant to the user in _that_ repo -- `build-single` and `build-multiple` are the truth there. There is no situation where conversion is skipped.

But it is not unimanginable that this particular tool is used in other tool chains. Or that the NTS build is used without this tool, or the narrative generator for that matter. This would be made impossible when building a monolithic NTS toolchain.

The reason `reference.file` is not used is that it is a per-project file. `convert.to.json.file` collects all information across projects.